addons/stock/models/product.py

```diff
@@ -774,7 +771,7 @@ def action_open_product_lot(self):
     def action_product_tmpl_forecast_report(self):
         action = self.env.ref('stock.report_stock_quantity_action_product').read()[0]
         action['domain'] = [
-            ('product_id', 'in', self.product_variant_ids.ids),
+            ('product_id', 'in', self.with_context(active_test=False).product_variant_ids.ids),
```

Comment:
would add a filter to remove archived product without quantity. In order to avoid noisy data

Reply:
@amoyaux  OK. I'll remove product with forecast quantity is 0. and in the the action_open_quants() i'll remove products with on hand quantity is 0